src/pd/cluster.rs

```diff
@@ -191,35 +192,39 @@ impl Connection {
     }
 
     // Re-establish connection with PD leader in synchronized fashion.
-    pub fn reconnect(
+    pub async fn reconnect(
         &self,
-        old_cluster: &Cluster,
+        mut cluster_guard: RwLockWriteGuard<'_, Cluster>,
         interval: u64,
         timeout: Duration,
-    ) -> Result<Option<Cluster>> {
-        if self.last_update.read().unwrap().elapsed() < Duration::from_secs(interval) {
+    ) -> Result<()> {
+        if cluster_guard.last_connected.elapsed() < Duration::from_secs(interval) {
             // Avoid unnecessary updating.
-            return Ok(None);
+            return Ok(());
         }
```

Comment:
I don't think we need the interval and the check here. If it's a repeated reconnecting, it will be rejected by checking the last updated time. What do you think?

Reply:
> I don't think we need the interval and the check here. If it's a repeated reconnecting, it will be rejected by checking the last updated time. What do you think?

Yes, I think this logic is out of date, removed now.